### ISSUE TITLE
Missing Pal Modules

### ISSUE THREAD

**Issue by jake09o** [localcc/PalworldModdingKit#12]
I followed your guidelines and executed them, but the project did not open, accompanied by an error message like the image.
![dsf](https://github.com/localcc/PalworldModdingKit/assets/34004748/87a30182-87aa-49cb-a822-747fdf56b696)

**Comment by AllelujahNightingale** [comment]
Same. Except i got to the next dialog box popup right after that one after clicking yes and got this. 
![image](https://github.com/localcc/PalworldModdingKit/assets/83302597/2f5845a3-ca3a-48c9-914e-d097f2347065)

**Comment by localcc** [comment]
Could you send the logs from `Saved/Logs/Pal.log`

**Comment by YaibaToKen** [comment]
[Pal.log](https://github.com/localcc/PalworldModdingKit/files/14066241/Pal.log)

Having the same issue, here's the requested log file. Hopefully it's just me doing something dumb.

**Comment by AllelujahNightingale** [comment]
[Pal.log](https://github.com/localcc/PalworldModdingKit/files/14066307/Pal.log)

Here is my log as well

**Comment by localcc** [comment]
> [Pal.log](https://github.com/localcc/PalworldModdingKit/files/14066241/Pal.log)
> 
> 
> 
> Having the same issue, here's the requested log file. Hopefully it's just me doing something dumb.

Pull the repo again, there have been fixes addressing the linker issue

**Comment by YaibaToKen** [comment]
> > [Pal.log](https://github.com/localcc/PalworldModdingKit/files/14066241/Pal.log)
> > Having the same issue, here's the requested log file. Hopefully it's just me doing something dumb.
> 
> Pull the repo again, there have been fixes addressing the linker issue

Project is now loading all the way! Ty!!

**Comment by MoRanYue** [comment]
I have encountered this problem, i cant open the project now.

**Comment by localcc** [comment]
Okay it seems it was not fixed, a fix is coming soon

**Comment by AllelujahNightingale** [comment]
As soon as my comment went up localcc posted a reply to the guy above saying it was fixed. It did work for me for the fix that they posted. So im finally in the editor.

**Comment by MoRanYue** [comment]
> As soon as my comment went up localcc posted a reply to the guy above saying it was fixed. It did work for me for the fix that they posted. So im finally in the editor.

I am waiting for the fixes online, and retry to opening it again and again ( )

**Comment by MoRanYue** [comment]
Through the log file, I found that it seemed it uses an hidden VS2019 to build these modules. Now i change build tools to VS2022 which I have installed just now, I'm observing the log now after I let it start to compile cpp files.

**Comment by localcc** [comment]
I cannot reproduce the linking issue with a fresh clone of the repo, which commit are you on? You can check that by running `git log` in your terminal in the sdk directory, if you have git installed

**Comment by localcc** [comment]
If you have `Source/Pal/Public/Pal.h` or `Source/Pal/Private/Pal.cpp`, delete those files

**Comment by MoRanYue** [comment]
> I cannot reproduce the linking issue with a fresh clone of the repo, which commit are you on? You can check that by running `git log` in your terminal in the sdk directory, if you have git installed

I am using the latest commit. by the way, i have been in the editor now. i applogize, it's actually my probelm

**Comment by WalterWoshid** [comment]
I just had the same problem, I think the solution is to fully close Unreal Engine after modifying `BuildConfiguration.xml`, had it open and received the same error.

**Comment by WalterWoshid** [comment]
Just had the problem, I had the editor open and didn't close it after editing the `BuildConfiguration.xml` file. After changing it, it still didn't work, but after waiting about 5 minutes and opening it again, it worked for some reason 🤔

**Comment by MoRanYue** [comment]
> Just had the problem, I had the editor open and didn't close it after editing the `BuildConfiguration.xml` file. After changing it, it still didn't work, but after waiting about 5 minutes and opening it again, it worked for some reason 🤔

At that time, I found it will write the log informations into file `<Project>/Saved/Logs/Pal.log`. Though I have already tried it again and again.

After I attempts to open the project again, it still gave me a window and it is like the picture of the first comment above. Then, I used VSCode to observe that log file realtimely.

When I clicked "Yes" on that msgbox, the log file said that it has started to build these missing modules, but it used a hidden VS2019 Buildtools in my computer, so.it didn't work. I edited `BuildConfiguration.xml` and ended up the building process. Then, I opened it, I turned to the log file, it said that it has started to build with a correct VS version.

It compiled 193 files, after it ends, the **HOLY**( ) Editor has appeared there finally!!!

**Comment by jordachemac** [comment]
I think I am having this same issue, using the latest commit 
![image](https://github.com/localcc/PalworldModdingKit/assets/455292/03c629d6-6e1a-4ae7-ad2d-7711ed41f9ca)
[Pal-backup-2024.01.27-22.42.50.log](https://github.com/localcc/PalworldModdingKit/files/14074218/Pal-backup-2024.01.27-22.42.50.log)

**Comment by MoRanYue** [comment]
> I think I am having this same issue, using the latest commit 
> ![image](https://github.com/localcc/PalworldModdingKit/assets/455292/03c629d6-6e1a-4ae7-ad2d-7711ed41f9ca)
> [Pal-backup-2024.01.27-22.42.50.log](https://github.com/localcc/PalworldModdingKit/files/14074218/Pal-backup-2024.01.27-22.42.50.log)
> 
> I have also set my VSCode version to 2022 in the BuildConfiguration.xml as per @MoRanYue comment yesterday but no luck
> 
> EDIT: Managed to get it working. After checking the log I realised it wasn't using VS2022 so I created an empty unreal project, went into editor preferences > source code. It was already set as VS2022 but I clicked "Set as default" and it gave me a warning that the settings file was unwriteable and would I like to make it writeable so I did and it set as default, then I was able to open the project. So unreal must be locking a version of the buildconfiguration file which stops it from recognising the changes entered manually. 

Yeah. When the project cant be opened, most of us had better check the log after letting it start to rebuild, if there is text that is like "VS2019", just change `BuildConfiguration.xml`.

**Comment by BraveJellybeanStudios** [comment]
I'd like to add my packing/cooking issues with the modkit were actually caused by this apparently. @jordachemac 's solution fixed my issues =D

**Comment by Sarfflow** [comment]
I have the same problem, but it seems to be different from others. 
[Pal.log](https://github.com/localcc/PalworldModdingKit/files/14080056/Pal.log)
It dose use VS2022.

I follow the instruction but something unexpected happened. 
![image](https://github.com/localcc/PalworldModdingKit/assets/101441863/a7482921-03bd-4f5e-8121-02343d2dbff7)
These folders in SDK are prefixed with UWP. I renamed them to fit the tutorial, but doesn't work. Could this be the cause of the problem?

**Comment by leesemin89** [comment]
[Pal.log](https://github.com/localcc/PalworldModdingKit/files/14190172/Pal.log)
Hi, I have same issue about this.. here is my logs.. 
I used only VS2022 and I set my unreal engine use VS2022, but it didn't work.

**Comment by giuliano-macedo** [comment]
I'm on the latest branch and getting some compille errors like:
```
error C2039: 'bReverseDirection': is not a member of 'UPalWorkFacingComponent'
```
[Pal.log](https://github.com/localcc/PalworldModdingKit/files/14202658/Pal.log)

**Comment by giuliano-macedo** [comment]
Thanks! It solved the compiler error, but now I'm getting this linker error:
```
error LNK2001: unresolved external symbol "public: virtual void __cdecl UPalWorkFacingComponent::GetLifetimeReplicatedProps(class TArray<class FLifetimeProperty,class TSizedDefaultAllocator<32> > &)const " (?GetLifetimeReplicatedProps@UPalWorkFacingComponent@@UEBAXAEAV?$TArray@VFLifetimeProperty@@V?$TSizedDefaultAllocator@$0CA@@@@@@Z)
```
[Pal.log](https://github.com/localcc/PalworldModdingKit/files/14208252/Pal.log)

**Comment by localcc** [comment]
> Thanks! It solved the compiler error, but now I'm getting this linker error:
> 
> ```
> 
> error LNK2001: unresolved external symbol "public: virtual void __cdecl UPalWorkFacingComponent::GetLifetimeReplicatedProps(class TArray<class FLifetimeProperty,class TSizedDefaultAllocator<32> > &)const " (?GetLifetimeReplicatedProps@UPalWorkFacingComponent@@UEBAXAEAV?$TArray@VFLifetimeProperty@@V?$TSizedDefaultAllocator@$0CA@@@@@@Z)
> 
> ```
> 
> [Pal.log](https://github.com/localcc/PalworldModdingKit/files/14208252/Pal.log)
> 
> 

Try pulling the latest commit again and seeing if that fixed it

**Comment by giuliano-macedo** [comment]
It worked, thanks!

**Comment by leesemin89** [comment]
Any suggestion for me? I thought I followed same instruction..

**Comment by giuliano-macedo** [comment]
@leesemin89 Your logs shows these errors:
* https://learn.microsoft.com/en-us/cpp/error-messages/compiler-errors-1/fatal-error-c1076?view=msvc-170
* https://learn.microsoft.com/en-us/cpp/error-messages/compiler-errors-2/compiler-error-c3859?view=msvc-170

Looking at this SO answer:
https://stackoverflow.com/a/50365521/5133524

It seems that the issue can be randomly caused in windows while under load, try closing any heavy RAM usage program and try to open again.

I think this can explain the intermittent issue that @WalterWoshid got [here](https://github.com/localcc/PalworldModdingKit/issues/12#issuecomment-1912726232)

**Comment by giuliano-macedo** [comment]
Hmm something mentioned in the SO:
> It could also be a Pagefile size problem (most likely on Virtual machines) but I believe you would have a message similar >to this :

>    c1xx : error C3859: Failed to create virtual memory for PCH [...Project.vcxproj] c1xx: note: the system returned code > 1455: The paging file is too small for this operation to complete

your log file does include the C3859 + 1455 error, are you in a VM?

**Comment by leesemin89** [comment]
@giuliano-macedo 
Thank you for your answer, I installed the Unreal engine and the project files on my HDD, Not SSD. 
My desktop has 32G DDR4.  Do you think HDD affects this project? 
- I am not in VM.

**Comment by giuliano-macedo** [comment]
I don't think it is a hard drive issue, I too installed the Unreal Engine on my HDD, and I have 16GB of RAM, other thing mentioned in the microsoft article and SO is that 32 bit VS can cause this issue, but in the log file it appears that you are using the 64 bit version... I think double check if you only have 64 bit VS would be good

**Comment by giuliano-macedo** [comment]
Another thing could be this:
> This one is easily overlook, but those kind of problem also happen when the precompiled header is just too big. Doing a little cleanup might be a good idea as well.

Try deleting the intermediate folder in the root directory of the PDK and try to open it again

**Comment by leesemin89** [comment]
@giuliano-macedo 
Good new! I just deleted `Intermediate` folder and tried rebuild again, then it sucessfully compiled.
Thank you for your help.

**Comment by Alicehaha6** [comment]
[Pal.log](https://github.com/localcc/PalworldModdingKit/files/14258631/Pal.log)
pleaze help me,I have same question

**Comment by RaccoonCaffeine** [comment]
Hi, i have problem starting up the proyect, i try to clone and repeat the procces multiple times without succes.
[Pal.log](https://github.com/user-attachments/files/16201024/Pal.log)

**Comment by RaccoonCaffeine** [comment]
Hi, i have a problem starting the project, i try clone the repo multiple times and the solutions above, but nothing seems to work for me
[Pal.log](https://github.com/user-attachments/files/16201154/Pal.log)

**Comment by Ghost7134** [comment]
Salut j'ai le meme probleme
[Pal.log](https://github.com/user-attachments/files/16240549/Pal.log)

**Comment by Ghost7134** [comment]
![Capture d'écran 2024-07-16 022110](https://github.com/user-attachments/assets/5deee48a-9590-44fa-87d7-c0f2661313b5)
![Capture d'écran 2024-07-15 234343](https://github.com/user-attachments/assets/bb6d3b57-ff0e-409f-bc97-c096d435b25a)
